sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -181,7 +178,7 @@ public EventHubProducerClient(string connectionString) : this(connectionString,
             OwnsConnection = true;
             Connection = new EventHubConnection(connectionString, eventHubName, clientOptions.ConnectionOptions);
             RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
-            EventHubProducer = Connection.CreateTransportProducer(null, RetryPolicy);
+            PartitionProducerPool = new TransportProducerPool(Connection.CreateTransportProducer(null, RetryPolicy));
```

Comment:
@jsquire 

I wonder if you think we should delegate the connection to return the pool?

Reply:
Not sure that I'm following what you mean.   Are you suggesting that we pass in a factory function for creating a transport producer for a given partition, rather than having the caller provide one when requesting a producer?

Reply:
My apologies my comment was quite poor.

Like having: ```Connection.CreateTransportProducerPool(null, RetryPolicy)```

Reply:
Since the pool is very specific to this scenario, I'd prefer to keep knowledge of it to just the producer client.   If we see it used elsewhere, that may be a conversation to have.    

One thing that I am noticing in the snippet.... maybe we should pass the connection into the pool and the retry policy;  they shouldn't change in the lifespan of the pool, since the pool is tied to the producer.   That would allow the pool to take responsibility for creating its own items.   It's a niche enough class that if we need to inject a specific transport producer we can always do that by mocking the connection.   

I'm thinking from this:
```csharp
PartitionProducerPool = new TransportProducerPool(Connection.CreateTransportProducer(null, RetryPolicy));
```

To this:
```csharp
PartitionProducerPool = new TransportProducerPool(Connection, RetryPolicy);
```

That may be useful in reducing complexity in allowing the pool to spawn transport producers without having to take the connection and retry policy multiple times.   _(or it may have been long enough that I'm forgetting some detail and this makes no sense, in which case ignore me)_